src/display/text_layer.js

```diff
@@ -585,7 +594,7 @@ var renderTextLayer = (function renderTextLayerClosure() {
         for (i = 0, ii = this._textDivs.length; i < ii; i++) {
           div = this._textDivs[i];
           div.style.padding = 0;
-          transform = div.dataset.originalTransform || '';
+          transform = divProperties.originalTransform || '';
```

Comment:
`divProperties` isn't defined in the `else` branch, which probably explains why the text layer isn't reset properly on mouseup.